### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'replace'

### ISSUE THREAD

**Issue by nsheff** [databio/peppro#12]
```PROCESSED=/project/shefflab/processed looper run ppqc/ppqc_config.yaml --selector-attribute sample_name --selector-include GSM3309957 --compute singularity_local
```
gives this:
```
Command completed. Elapsed time: 0:00:00. Running peak memory: 0.001GB.
  PID: 32;      Command: ln;    Return code: 0; Memory used: 0.001GB
Local input file: '/project/shefflab/processed/ppqc/results_pipeline/GSM3309957/raw/GSM3309957.bam'
Found .bam file
Traceback (most recent call last):
  File "/home/ns5bc/code/peppro/pipelines/peppro.py", line 3003, in <module>
    sys.exit(main())
  File "/home/ns5bc/code/peppro/pipelines/peppro.py", line 1649, in main
    container=pm.container)
  File "/usr/local/lib/python3.6/dist-packages/pypiper/manager.py", line 641, in run
    lock_name = lock_name or make_lock_name(target, self.outfolder)
  File "/usr/local/lib/python3.6/dist-packages/pypiper/utils.py", line 515, in make_lock_name
    return [make_name(p) for p in original_path]
  File "/usr/local/lib/python3.6/dist-packages/pypiper/utils.py", line 515, in <listcomp>
    return [make_name(p) for p in original_path]
  File "/usr/local/lib/python3.6/dist-packages/pypiper/utils.py", line 511, in make_name
    return p.replace(path_base_folder, "").replace(os.sep, "__")
AttributeError: 'NoneType' object has no attribute 'replace'
Pipeline status: running
Starting cleanup: 0 files; 0 conditional files for cleanup
```

having trouble finding the source of this error

**Comment by jpsmith5** [comment]
Looks like this is related to the combination of `pypiper`'s `ngstk`'s `bam_to_fastq_awk` function returning `None` for read2 of single end data.

```
if paired_end:
    cmd = self.tools.samtools + " view " + bam_file + " | awk '"
    cmd += r'{ if (NR%2==1) print "@"$1"/1\n"$10"\n+\n"$11' + fq1_target + ';'
    cmd += r' else print "@"$1"/2\n"$10"\n+\n"$11' + fq2_target + '; }'
    cmd += "'"  # end the awk command
else:
    fq2 = None
    cmd = self.tools.samtools + " view " + bam_file + " | awk '"
    cmd += r'{ print "@"$1"\n"$10"\n+\n"$11' + fq1_target + '; }'
    cmd += "'"
return cmd, fq1, fq2
```

That in conjunction with `pypiper`'s `utils` `make_lock_name` not being able to handle `None` type.

**Comment by nsheff** [comment]
This is fixed in pypiper 0.12.0.
